### DISCUSSION TITLE
Get the value of the selected option in Muliselect

### DISCUSSION THREAD

**Discussion by mattismoel** [chhoumann/quickadd#508] [Q&A]
Is it in any way possible to retrieve the name of QuickAdd selection choices, like a {{CHOICE}} tag or similar? I am trying to have this general template for lecture notes, where it connects with a link to the main note of that class. I would much prefer to have the same base template for all these lessons, where this link is just replaced with the choice of class that I want to take notes on. Is this possible?

I basically want to be able to select "English", "History" etc. from the QuickAdd menu, and then in my BaseLecture template it will do:
Class: [[CHOICE]].

Hope it makes sense

**Comment by nicobako** [comment]
Have you tried creating an [inline script](https://quickadd.obsidian.guide/docs/InlineScripts)? That would be my suggestion.

I think what would work for you is an inline script where you `await` the results of a [suggester prompt](https://quickadd.obsidian.guide/docs/QuickAddAPI#suggester-displayitems-string--value-string-index-number-arr-string--string-actualitems-string-promisestring).

Let me know if this makes sense, or if you need further clarification.

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

This is built into the format syntax now - no script needed. A named `{{VALUE}}` with comma-separated options opens a suggester instead of a text prompt, so your base lecture template can contain:

```
Class: [[{{VALUE:English,History,Math}}]]
```

Running the choice pops a pick-list with those options, and the one you choose is inserted - and reused anywhere the same token appears in that run.

Useful extras:

- `{{VALUE:English,History,Math|custom}}` also lets you type a value that isn't in the list.
- `|label:Pick a class` adds helper text to the suggester.

@nicobako's inline-script suggestion still makes sense when the option list has to be computed dynamically (say, from the files in a Classes folder) - but for a fixed set of classes the token above is all you need.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#named-value
